### PR TITLE
Feat/search string cleanup

### PR DESCRIPTION
**Pull request by aseiden**
tl;dr: Cleans data in slightly different ways prior to aylien search and db insert.

index.jsx: Became necessary to slightly modify the server response to a search to include the search term so the graph displayed the properly cleaned search term.  Changed this file to adjust for the new data structure.

server.js: Runs data through cleaning helpers prior to passing them off to the aylien/db insert functions.

cleanSearch: Helper file with 3 separate cleaning functions and 1 additional internal helper.

makeTimeline.js: Modified to pass along necessary data for proper front-end display.

screen shot 2017-05-11 at 7 43 04 pm

### CODE REVIEW

File: client/app/index.jsx [hunk 1]
```diff
@@ -34,9 +34,9 @@ class App extends React.Component {
       params: { q: trend }
     })
     .then(response => {
-      let timeline = response.data;
+      let timeline = response.data.timeline;
       this.setState({
-        trend: trend,
+        trend: response.data.trend,
```
Comment (epan) [inline client/app/index.jsx:39]: omg ive wanted to capitalize this for the chart title. not sure if better here or at the chart component

File: utilities/cleanSearch.js [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+    }
+  }
+
+  return true;
+};
+
+const prepForAylien = (trend) => {
+  trend = trend.toLowerCase();
+
+  if (trend.length === 0 || checkForAllSpaces(trend)) {
```
Comment (epan) [inline utilities/cleanSearch.js]: u prob already know this but another option for `if` `trend === null` and `trend.length === 0` is `!trend` and `!trend.length` 😃

File: utilities/cleanSearch.js [hunk 3]
```diff
@@ -0,0 +1,45 @@
+const checkForAllSpaces = (trend) => {
```
Comment (epan) [inline utilities/cleanSearch.js:1]: for functions that return `true` or `false` like this I personally write the names like `isAllSpaces` because the answer is `true` or `false` or `hasNumber` instead of `checkForNumber` no need to change, just thought I'd share if you find it useful

Comment (epan) [reply utilities/cleanSearch.js:1]: `if (trend.length === 0 || checkForAllSpaces(trend)) {`
vs
`if (trend.length === 0 || isAllSpaces(trend)) {`

File: server.js [hunk 4]
```diff
@@ -58,5 +70,5 @@ app.get('/api/history', (req, res) => {
 app.use((req, res) => {
   res.status(404);
   res.sendFile(__dirname + '/404.html');
-})
+});
```
Comment (epan) [inline server.js:73]: lol nice

**Review by epan — APPROVED** [review]
cool ❄️
